src/modalities/dataloader/create_packed_data.py

```diff
@@ -312,7 +323,10 @@ def _process_line(self, line: str, process_id: int) -> bytes:
         tokens = self.tokenizer.tokenize(jq_retrieved_text)
         if len(tokens) == 0:
             raise EmptySampleError("Received empty sample...")
-        return b"".join(map(self._encoded_token_to_bytes, tokens)) + self._encoded_eos_token_as_bytes
+        token_byte_string = b"".join(map(self._encoded_token_to_bytes, tokens))
+        if not token_byte_string.endswith(self._encoded_eos_token_as_bytes):
```

Comment:
```suggestion
        if not token_byte_string.endswith(self._encoded_eod_token_as_bytes):
```
Use `eod` consistently instead of both `eos` and `eod`